engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/DynamicGlobalVariable.scala

```diff
@@ -0,0 +1,13 @@
+package pl.touk.nussknacker.engine.api.typed
+
+import pl.touk.nussknacker.engine.api.MetaData
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+
+/**
+  * Trait to be implemented by global variable to provide dynamic value and more detailed typing info.
+  */
+trait DynamicGlobalVariable {
+  def value(metadata: MetaData): Any
+  def returnType(metadata: MetaData): TypingResult
+  def runtimeClass: Class[_]
```

Comment:
I think it's used not as runtime class, but to provide return type for "base" use case, where there is no metadata yet? 
Let's rename it and let it return TypedResult?

Reply:
You're right. Done.